src/store/authentication/saga.ts

```diff
@@ -28,90 +30,75 @@ export const currentUserSelector = () => (state) => {
   return getDeepProperty(state, 'authentication.user.data', null);
 };
 
-export function* nonceOrAuthorize(action) {
-  yield processUserAccount({ user: null, nonce: null, chatAccessToken: null, isLoading: true });
+export function* setAuthentication({ chatAccessToken } = { chatAccessToken: '' }) {
+  yield put(setChatAccessToken({ value: chatAccessToken, isLoading: false }));
+}
 
+export function* nonceOrAuthorize(action) {
   const { signedWeb3Token } = action.payload;
-
   const { nonceToken: nonce = undefined, chatAccessToken } = yield call(nonceOrAuthorizeApi, signedWeb3Token);
-
   if (nonce) {
-    yield put(setUser({ nonce, isLoading: false }));
+    yield put(setUser({ nonce, data: null }));
   } else {
-    const user = yield call(fetchCurrentUser);
-
-    yield processUserAccount({ user, chatAccessToken, isLoading: false });
+    yield setAuthentication({ chatAccessToken });
+    yield call(completeUserLogin);
   }
 
   return { nonce };
 }
 
-export function* terminate() {
-  yield processUserAccount({ user: null, nonce: null, chatAccessToken: null, isLoading: false });
+export function* completeUserLogin(user = null) {
+  if (!user) {
+    user = yield call(fetchCurrentUser);
```

Comment:
what happens if the `fetchCurrentUser` fails? Do we fail here, or return `null` & continue?

Reply:
yeah, agree. A global error state for handling an error while authentication makes sense to me